Userland/Libraries/LibWeb/FileAPI/File.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2022, Kenneth Myhra <kennethmyhra@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibWeb/FileAPI/File.h>
+
+namespace Web::FileAPI {
+
+static bool is_basic_latin(StringView view)
+{
+    for (auto code_point : view) {
+        if (code_point < 0x0020 || code_point > 0x007E)
+            return false;
+    }
+    return true;
+}
+
+File::File(ByteBuffer const& byte_buffer, String const& name, String const& type, i64 const last_modified)
+    : Blob(byte_buffer, type)
+    , m_name(name)
+    , m_last_modified(last_modified)
+{
+}
+
+// https://w3c.github.io/FileAPI/#ref-for-dom-file-file
+DOM::ExceptionOr<NonnullRefPtr<File>> File::create(Vector<BlobPart> const& file_bits, String const& name, Optional<FilePropertyBag> const& options)
+{
+    // 1. Let bytes be the result of processing blob parts given fileBits and options.
+    auto bytes = TRY_OR_RETURN_OOM(process_blob_parts(file_bits));
+
+    // 2. Let n be the fileName argument to the constructor.
+    //    NOTE: Underlying OS filesystems use differing conventions for file name; with constructed files, mandating UTF-16 lessens ambiquity when file names are converted to byte sequences.
+    //    NOTE: Treating this as a no-op since we pass the name argument directly to the File constructor in step 4
+
+    auto type = String::empty();
+    i64 last_modified = 0;
+    // 3. Process FilePropertyBag dictionary argument by running the following substeps:
+    if (options.has_value()) {
+
+        // 1. If the type member is provided and is not the empty string, let t be set to the type dictionary member.
+        //    If t contains any characters outside the range U+0020 to U+007E, then set t to the empty string and return from these substeps.
```

Comment:
> return from these substeps.

Reply:
This feels like maybe a spec issue? Seems strange to return from these substeps when step 3 of these substeps involves handling the `lastModified` member. Or maybe I'm not interpreting `return from these substeps` correctly

Reply:
Indeed, looks like a spec issue. At least firefox does step 3 unconditionally: https://searchfox.org/mozilla-central/source/dom/file/File.cpp#144

Reply:
...and so does Chromium: https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/fileapi/blob.cc;l=128;drc=40c209a9c365ebb9f16fb99dfe78c7fe768b9594

Reply:
Since we seem to more or less align with other browsers interpretation of the spec I'll set this to resolved.